pypeit/flatfield.py

```diff
@@ -1317,10 +1317,28 @@ def fit(self, spat_illum_only=False, doqa=True, debug=False):
             # Construct the pixel flat
             #trimmed_slitid_img_anew = self.slits.slit_img(pad=-trim, slitidx=slit_idx)
             #onslit_trimmed_anew = trimmed_slitid_img_anew == slit_spat
-            self.mspixelflat[onslit_tweak] = rawflat[onslit_tweak]/self.flat_model[onslit_tweak]
+            self.mspixelflat[onslit_tweak] = rawflat[onslit_tweak] * utils.inverse(self.flat_model[onslit_tweak])
             # TODO: Add some code here to treat the edges and places where fits
             #  go bad?
 
+            # Check for infinities and NaNs
+            winfnan = np.where(np.logical_not(np.isfinite(self.flat_model[onslit_tweak])))
+            if winfnan[0].size != 0:
+                msgs.warn('There are {0:d} pixels with non-finite values in the flat-field model '
+                          'for slit {1:d}!'.format(winfnan[0].size, slit_spat) + msgs.newline() +
+                          'These model pixel values will be set to the raw pixel value.')
+                self.flat_model[np.where(onslit_tweak)[0][winfnan]] = rawflat[np.where(onslit_tweak)[0][winfnan]]
+                self.mspixelflat[np.where(onslit_tweak)[0][winfnan]] = 1.0
```

Comment:
I feel like we should be masking these instead, but it's likely a lot of work to propagate the masking through the code...